simplefi-balancer-v2/src/vault.ts

```diff
@@ -0,0 +1,183 @@
+import { 
+  Address, 
+  BigInt, 
+  ethereum, 
+  log,
+} from "@graphprotocol/graph-ts"
+
+import {
+  PoolRegistered,
+  TokensDeregistered,
+  TokensRegistered,
+  PoolBalanceChanged,
+} from "../generated/Vault/Vault"
+
+import {
+  getOrCreateAccount,
+  getOrCreateERC20Token,
+  getOrCreateMarket,
+  updateMarket,
+  TokenBalance,
+} from "./common"
+
+import {
+  Pool as PoolEntity,
+  PoolId as PoolIdEntity,
+  Token as TokenEntity,
+  Mint as MintEntity,
+  Burn as BurnEntity,
+  Market as MarketEntity,
+} from "../generated/schema"
+
+import { 
+  ProtocolName, 
+  ProtocolType,
+  PoolSpecialization
+} from "./constants"
+
+import {
+  createOrUpdatePositionOnMint,
+  createOrUpdatePositionOnBurn
+} from "./market"
+
+export function handlePoolRegistered(event: PoolRegistered): void {
+  let pool = new PoolEntity(event.params.poolId.toHexString())
+  pool.address = event.params.poolAddress.toHexString();
+
+  let evmEvent = event as ethereum.Event
+  pool.blockNumber = evmEvent.block.number
+  pool.timestamp = evmEvent.block.timestamp
+  // pool.poolSpecialization = event.params.specialization // PoolSpecialization.GENERAL
+  pool.save()
+
+  // Solely purpose to retrive poolId from poolAddress when needed
+  let poolId = new PoolIdEntity(pool.address)
+  poolId.poolId = pool.id
+  poolId.save()
+}
+
+export function handleTokensRegistered(event: TokensRegistered): void {
+  let pool = PoolEntity.load(event.params.poolId.toHexString())
+
+  // @todo: what if the pool already existed? (more tokens added)
+  // Create a tokens and market entity
+  let tokens: TokenEntity[] = []
+  let tokensStr: string[] = [] // @todo: Is there any workaround to avoid 2 arrays?
```

Comment:
@NelsonGaldeman you don't need to create this second array for strings you can simply use `tokens.map(t => t.id)` to assing to `pool.tokens`